compiler/AST/primitive.cpp

```diff
@@ -702,11 +702,12 @@ initPrimitive() {
   prim_def(PRIM_REQUIRE, "require", returnInfoVoid, false, false);
 }
 
-Map<const char*, VarSymbol*> memDescsMap;
+static Map<const char*, VarSymbol*> memDescsMap;
```

Comment:
The simple solution to that is to give newMemDesc a special case for string symbols. If you think that is better than having 2 maps (I do, but it's not a sure thing) then it's worth trying.